File: Coding Exercises/Python/Day34/distribute_chocolates.py
def distribute_chocolates(chocolates, num_children):
    num_chocolates = len(chocolates)
    chocolates_per_child = num_chocolates // num_children
    distribution = [chocolates_per_child] * num_children
    chocolates_left = num_chocolates % num_children
    for i in range(chocolates_left):
        distribution[i] += 1
    return distribution

File: Coding Exercises/TypeScript/Day34/distributeChocolates.ts
function distributeChocolates(chocolates: number[], numChildren: number): number[] {
  const numChocolates: number = chocolates.length;
  const chocolatesPerChild: number = Math.floor(numChocolates / numChildren);
  const distribution: number[] = Array(numChildren).fill(chocolatesPerChild);
  const chocolatesLeft: number = numChocolates % numChildren;
  for (let i = 0; i < chocolatesLeft; i++) {
      distribution[i]++;
  }
  return distribution;
}
